Kernel/Tasks/Thread.cpp

```diff
@@ -1264,6 +1240,9 @@ void Thread::set_state(State new_state, u8 stop_signal)
     } else if (m_state == Thread::State::Stopped) {
         // We don't want to restore to Running state, only Runnable!
         m_stop_state = previous_state != Thread::State::Running ? previous_state : Thread::State::Runnable;
```

Comment:
Shouldn't this now always be `m_stop_state = Runnable` and therefore make `m_stop_state` unnecessary?